### PR TITLE
:sparkles: add --ignore-errors flag

### PR DESCRIPTION
**Pull request by CWBudde**
Along with the --ignore-errors flag I also implemented a couple of other minor fixes. In the end it led to the fact that only 4 errors (due to bad design in the original sqlite3 database) needed to get fixed manually.

In the end I'm not 100% sure that the current code (after several revisions) is the best it could be. So I'd be happy about a critical review. In case the changes are too much, I could split this PR.

### CODE REVIEW

File: sqlite3_to_mysql/sqlite_utils.py [hunk 1]
```diff
@@ -41,7 +41,11 @@ def unicase_compare(string_1: str, string_2: str) -> int:
 def convert_date(value) -> date:
     """Handle SQLite date conversion."""
     try:
-        return date.fromisoformat(value.decode())
+        decoded_value: str = value.decode()
+        if " " in decoded_value:
+            decoded_value = decoded_value.split(" ")[0]
```
Comment (techouse) [inline sqlite3_to_mysql/sqlite_utils.py:46]: This smells to me like you're trying to mange a datetime into a date. Not sure I like it.

File: sqlite3_to_mysql/transporter.py [hunk 2]
```diff
@@ -251,7 +252,7 @@ def _translate_type_from_sqlite_to_mysql(self, column_type: str) -> str:
         full_column_type: str = column_type.upper()
         match: t.Optional[t.Match[str]] = self._valid_column_type(column_type)
         if not match:
-            raise ValueError("Invalid column_type!")
+            raise ValueError("Invalid column_type: " + column_type)
```
Comment (techouse) [inline sqlite3_to_mysql/transporter.py:255]: Please use an f-string here, i.e.

```python
raise ValueError(f"Invalid column_type: {column_type}")
```

File: sqlite3_to_mysql/transporter.py [hunk 3]
```diff
@@ -330,13 +340,15 @@ def _create_table(self, table_name: str, transfer_rowid: bool = False) -> None:
...
+            not_null: bool = column["notnull"] or column["pk"]
+
             sql += " `{name}` {type} {notnull} {default} {auto_increment}, ".format(
                 name=mysql_safe_name,
                 type=column_type,
-                notnull="NOT NULL" if column["notnull"] or column["pk"] else "NULL",
+                notnull="NOT NULL" if not_null else "NULL",
                 auto_increment="AUTO_INCREMENT" if auto_increment else "",
-                default="DEFAULT " + column["dflt_value"]
-                if column["dflt_value"] and column_type not in MYSQL_COLUMN_TYPES_WITHOUT_DEFAULT and not auto_increment
```
Comment (techouse) [inline sqlite3_to_mysql/transporter.py:339]: So here you want to completely ignore the fact that a field can not have a default type? Why?

File: sqlite3_to_mysql/transporter.py [hunk 4]
```diff
@@ -330,13 +340,15 @@ def _create_table(self, table_name: str, transfer_rowid: bool = False) -> None:
                 column["pk"] > 0 and column_type.startswith(("INT", "BIGINT")) and not compound_primary_key
             )
 
+            not_null: bool = column["notnull"] or column["pk"]
+
             sql += " `{name}` {type} {notnull} {default} {auto_increment}, ".format(
                 name=mysql_safe_name,
                 type=column_type,
-                notnull="NOT NULL" if column["notnull"] or column["pk"] else "NULL",
+                notnull="NOT NULL" if not_null else "NULL",
```
Comment (techouse) [inline sqlite3_to_mysql/transporter.py:348]: I prefer the explicit condition.

**Comment by techouse** [comment]
I've left a few comments. Can you please take a look at them.

Also, please add unit tests.

File: sqlite3_to_mysql/sqlite_utils.py (hunk 1 above)
Comment (CWBudde) [reply sqlite3_to_mysql/sqlite_utils.py:46]: Indeed. I'm sorry that I forgot to remove this piece and reverted that code. It was a bad and legacy database I tried to convert.

File: sqlite3_to_mysql/transporter.py (hunk 4 above)
Comment (CWBudde) [reply sqlite3_to_mysql/transporter.py:348]: OK

**Comment by techouse** [comment]
@CWBudde I closed this PR because it hasn't seen any activity in over 6 months. Please feel free to reopen if needed.

**Pull request closed without merging**